src/provider_completions.jl

```diff
@@ -1,7 +1,23 @@
 function process(r::JSONRPC.Request{Val{Symbol("textDocument/completion")},TextDocumentPositionParams}, server)
     tdpp = r.params
     line = get_line(tdpp, server)
-    comp = Base.REPLCompletions.completions(line, chr2ind(line,tdpp.position.character))[1]
+    io = IOBuffer(reverse(line[1:chr2ind(line,tdpp.position.character)]))
+    word = Char[]
+    while !eof(io)
+        c = read(io, Char)
+        if c=='\\'
+            unshift!(word, c)
+            break
+        end
+        if !(Base.is_id_char(c) || c=='.' || c=='_' || c=='^')
+            break
+        end
+        unshift!(word, c)
+    end
+    
+    str = String(word)
+    prefix = str[1:findlast(word,'.')]
+    comp = Base.REPLCompletions.completions(str,length(str))[1]
```

Comment:
``length`` is incorrect, because you need to pass the byte count here, not the character count, I believe.